apps/src/gamelab/spritelab/spriteCommands.js

```diff
@@ -0,0 +1,98 @@
+import * as spriteUtils from './spriteUtils';
+
+export const commands = {
+  countByAnimation(animation) {
+    let sprites = spriteUtils.singleOrGroup(animation);
+    return sprites.length;
+  },
+  destroy(spriteId) {
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    sprites.forEach(sprite => {
+      sprite.destroy();
+      spriteUtils.removeAllBehaviors(sprite);
+      spriteUtils.deleteSprite(sprite.id);
+    });
+  },
+
+  displace(spriteId, targetSpriteIndex) {
+    let sprites = spriteUtils.singleOrGroup(spriteId);
+    let targetSprites = spriteUtils.singleOrGroup(targetSpriteIndex);
+    sprites.forEach(sprite => {
+      targetSprites.forEach(target => sprite.displace(target));
+    });
+  },
+
+  getProp(spriteId, prop) {
+    let sprite = spriteUtils.singleOrGroup(spriteId)[0];
+    if (sprite !== undefined) {
+      if (prop === 'scale') {
+        return sprite.getScale() * 100;
+      } else if (prop === 'costume') {
+        return sprite.getAnimationLabel();
+      } else if (prop === 'y') {
+        return 400 - sprite.y;
+      } else {
+        return sprite[prop];
+      }
+    }
+  },
+
+  getThisSprite(which, extraArgs) {
+    if (extraArgs) {
+      if (which === 'this') {
+        return extraArgs.sprite;
+      }
+      if (which === 'other') {
+        return extraArgs.target;
+      }
+    }
+  },
+
+  makeSprite(animation, location) {
+    if (!location) {
+      location = {x: 200, y: 200};
+    }
+    if (typeof location === 'function') {
+      location = location();
+    }
+    var sprite = this.createSprite(location.x, location.y);
+    sprite.direction = 0;
+    sprite.baseScale = 1;
+    sprite.setScale = function(scale) {
+      sprite.scale = scale * sprite.baseScale;
+    };
+    sprite.getScale = function() {
+      return sprite.scale / sprite.baseScale;
```

Comment:
Can sprite.baseScale ever be 0? Do we need to handle a case where baseScale may not exist?

Reply:
baseScale is guaranteed to exist (it's initialized to 1 in `makeSprite()`), and the only time it's reassigned, it's set to `100 / Math.max(100, height, width)`, so also guaranteed to be not 0.

Reply:
👍